utam-compiler/src/main/java/utam/compiler/grammar/UtamMethodAction.java

```diff
@@ -145,7 +150,7 @@ ComposeMethodStatement getComposeAction(TranslationContext context, MethodContex
       );
     }
     if (apply != null) {
-      if (elementName == null) {
+      if (!isWaitFor(apply) && elementName == null) {
```

Comment:
I understand your point of view and it seems that this feature is a bit confusing (self vs root vs elements and the divergences in compiler implementations). However, the [RFC states](https://rfcs.lwc.dev/rfcs/utam/2021-02-26-self):

> The subject of this RFC is to support invocations of the current page objects methods from a compose statement by using `"element":"self"` to represent the current page object. **It will also be the default value if no element is set.**

I propose that we implement the feature as specified by the RFC and we can suggest a new revision of the RFC after a team discussion if we don't really like the implicit fallback to `self`. I am mainly proposing that as that's what we have in JS and also what is documented (and published) on the doc site. Just to avoid any parity gap and then we can update both compilers to have only explicit value for compose statements elements.

Reply:
@olivier-martin-sf You're right, of course, about what the RFC says. I'd overlooked that statement in the spec, and that makes my comment inapplicable here. Given that, I'm okay with this change.

Reply:
I tried adding a self element, but the questions I had were around creating a pseudo element and it seemed simpler to stick just checking if the reference is the String "self". Would you think that would be a better way to handle this @jimevanssfdc ?

Reply:
to me omitting `"element":"self"` is totally fine, probably because i always thought that for waitFor it was redundant as it's an abstraction and not something applied to the element (only wrapped action is), but we can always change it later